src/main/java/com/wepay/zktools/zookeeper/internal/ZooKeeperClientImpl.java

```diff
@@ -55,6 +58,14 @@
     private volatile boolean running = true;
 
     public ZooKeeperClientImpl(String connectString, int sessionTimeout) throws ZooKeeperClientException {
+        this(connectString, sessionTimeout, null);
+    }
+
+    /**
+     * Avoid directly use this constructor with a not-null connectTimeout parameter, unless you have to timeout this
```

Comment:
nit: maybe here: unless you purposefully want to avoid Zktools to manage the zookeeper session lifecycle for you?

Reply:
Good suggestion 👍 Will update it.